sklearn/datasets/tests/test_openml.py

```diff
@@ -1190,3 +1191,44 @@ def test_fetch_openml_with_ignored_feature(monkeypatch, gzip_response):
     # so we assert that we don't have the ignored feature in the final Bunch
     assert dataset['data'].shape == (101, 16)
     assert 'animal' not in dataset['feature_names']
+
+
+@pytest.mark.parametrize('as_frame', [True, False])
+def test_fetch_openml_verify_checksum(monkeypatch, as_frame, cache):
+    data_id = 2
+    _monkey_patch_webbased_functions(monkeypatch, data_id, True)
+
+    # create a modified (local) arff file
+    dataset_dir = os.path.join(currdir, 'data', 'openml', str(data_id))
+    original_data_path = os.path.join(dataset_dir,
+                                      'data-v1-download-1666876.arff.gz')
+    corrupt_copy = original_data_path + ".test_corrupt_arff"
+    with gzip.GzipFile(original_data_path, "rb") as orig_gzip, \
+         gzip.GzipFile(corrupt_copy, "wb") as modified_gzip:
+        data = bytearray(orig_gzip.read())
+        data[len(data)-1] = 37
+        modified_gzip.write(data)
+
+    # simulate request to return modified file
+    mocked_openml_url = sklearn.datasets._openml.urlopen
```

Comment:
Leave a comment that `sklearn.datasets._openml.urlopen` is already mocked by `_monkey_patch_webbased_functions`.